giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/apps/IRegisteredApp.java

```diff
@@ -26,4 +26,13 @@
 
     public abstract String getProviderClientId();
 
+    /**
+     * authorization type required to use in token validation.
+     * @return empty string for IdToken validation
+     * @return non-empty string for accessToken validation
+     */
```

Comment:
this method should return the type of authorization used. doesn't have to be just one or the other. just say it return whatever the authorization type is. that by default it uses openid if nothing is specified does not belong here.